### PR TITLE
Add documentation on how to use Lens Extension Generator

### LINKED ISSUE #6: Document how to use generator in lens/doc
and fire PR to lens main repo.

### PR DESCRIPTION
**Pull request by chenhunghan**
Closes https://github.com/lensapp/generator-lens-ext/issues/6

Signed-off-by: Hung-Han (Henry) Chen <1474479+chenhunghan@users.noreply.github.com>

### CODE REVIEW

File: docs/extensions/get-started/generator.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+## Debugging the Extension
+
+[Testing](../testing-and-publishing/testing.md)
+
+## Next steps
+
+You can take a closer look at [Common Capabilities](../capabilities/common-capabilities.md) of extension, how to [style](../capabilities/styling.md) the extension. Or the [Extension Anatomy](anatomy.md).
+
+Welcome to to raise an [issue](https://github.com/lensapp/generator-lens-ext/issues) for Lens Extension Generator, if you find problems, or have feature requests.
```
Comment (panuhorsmalahti) [inline docs/extensions/get-started/generator.md]: You are welcome to raise

Comment (chenhunghan) [reply docs/extensions/get-started/generator.md]: fixed https://github.com/lensapp/lens/pull/1411/commits/1709502237dd8a16cef60df536c0992baffc298d

File: mkdocs.yml [hunk 2]
```diff
@@ -22,6 +22,7 @@ nav:
     - Getting Started:
       - Overview: extensions/get-started/overview.md
       - Your First Extension: extensions/get-started/your-first-extension.md
+      - Generator: extensions/get-started/generator.md
```
Comment (jakolehm) [inline mkdocs.yml]: Maybe this should be under `Extension Guides`?

File: docs/extensions/get-started/generator.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+```bash
+npm install -g yo generator-lens-ext
+```
+
+Run the generator and fill out a few fields for a TypeScript project:
+
+```bash
+yo lens-ext
+# ? What type of extension do you want to create? New Extension (TypeScript)
+# ? What's the name of your extension? my-first-lens-ext
```
Comment (Nokel81) [inline docs/extensions/get-started/generator.md]: At least when I run the generator the suggested name is "my-ne-lens-extension"

File: docs/extensions/get-started/generator.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+# ? What's the description of your extension? My hello world extension
+# ? What's your extension's publisher name? @my-org/my-first-lens-ext
```
Comment (Nokel81) [inline docs/extensions/get-started/generator.md]: Is it possible for a `yo` generator to validate this field? Make sure that the user is following the `@<pub_name>/<name>` convention?

Comment (chenhunghan) [reply docs/extensions/get-started/generator.md]: Good point, it's possible, I will open an issue in generator repo https://github.com/lensapp/generator-lens-ext/issues/7

File: docs/extensions/get-started/generator.md (hunk 3 above)
Comment (chenhunghan) [reply docs/extensions/get-started/generator.md]: Thanks, fixed a typo, sync the default to generator repo https://github.com/lensapp/generator-lens-ext/pull/8

File: mkdocs.yml (hunk 2 above)
Comment (chenhunghan) [reply mkdocs.yml]: It seems that extension guides is for introduction of main/renderer extensions, like this PR https://github.com/lensapp/lens/pull/1427/files If generator is for developer to bootstrap new projects, maybe better in the get started section?

Comment (jakolehm) [reply mkdocs.yml]: Still not sure about that. @nevalla @stevejr @msa0311 wdyt?

Comment (nevalla) [reply mkdocs.yml]: If generator produces working extension like VS Code generator does, then it's fine to have in get started section. But in our case we are cloning the repo containing all the bits and pieces so it's a bit redundant step there. But the generator might be mentioned in Next steps section, for example:

>  Extension Guides contain instruction to install extension generator tool and additional samples to get started with your own extension.

File: docs/extensions/get-started/generator.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+# ? Initialize a git repository? Yes
+# ? Install dependencies after initialization? Yes
+# ? Which package manager to use? yarn
+# ? symlink created extension folder to ~/.k8slens/extensions (mac/linux) or :User
+s<user>.k8slensextensions (win)? Yes
```
Comment (nevalla) [inline docs/extensions/get-started/generator.md]: This is more related to generator itself, but is the window path correct?

Comment (chenhunghan) [reply docs/extensions/get-started/generator.md]: Probably not... I copied from here https://docs.k8slens.dev/master/extensions/get-started/your-first-extension/#windows

And @nevalla do you know what's the path to home folder on Windows (`~/` on Mac/Linux)? Now I remember I haven't implemented symlink on Windows yet https://github.com/lensapp/generator-lens-ext/issues/10 

Found this
https://stackoverflow.com/questions/9228950/what-is-the-alternative-for-users-home-directory-on-windows-command-prompt

Comment (nevalla) [reply docs/extensions/get-started/generator.md]: okay probably it's just missing backslash. Correct should be `<user>\.k8slens\extensions` also on windows.

Comment (chenhunghan) [reply docs/extensions/get-started/generator.md]: Yeah... not sure what happens to my copy-paste, let's blame Big Sur. fixed in https://github.com/lensapp/lens/pull/1411/commits/1d763a1877937e0616179e936ccf9e1aa6a629a7

File: mkdocs.yml (hunk 2 above)
Comment (chenhunghan) [reply mkdocs.yml]: Ok, fixed in https://github.com/lensapp/lens/pull/1411/commits/5d904eac56b8bed0dda416411d9bfb6a8cae944c

**Review by jakolehm — APPROVED** [review]

**Pull request merged**